packages/earth-admin/src/pages-client/layers/details.tsx

```diff
@@ -14,133 +14,476 @@
+    mode: 'onChange',
+  });
+
+  const {touched, dirty, isValid} = formState;
+  const renderErrorFor = setupErrors(errors, touched);
 
   useEffect(() => {
-    async function setupLayers() {
-      setIsLoading(true);
+    setFormValid(isValid);
+  }, [isValid]);
+
 
-      const query = {
-        search: searchValue,
-        sort: 'name',
-        page: {size: pageSize, cursor: pageCursor},
-        group: selectedGroup,
-      };
-      const encodedQuery = encodeQueryToURL('layers', query);
-      const res: any = await getAllLayers(encodedQuery);
+  async function onSubmit(e?, setIsEditing?, setIsLoading?, setServerErrors?) {
+    e.preventDefault();
 
+    const formData = getValues();
 
-      setTotalResults(res.total);
+    const {type, category, provider, references} = formData;
 
-      setLayer(!nextCursor ? res.data : [...layers, ...res.data]);
-      setNextCursor(res.pagination.nextCursor ? res.pagination.nextCursor : null);
-      setIsNoMore(!res.pagination.nextCursor);
+    const parsed = {
+      ...formData,
+      ...(category && {category: flattenArrayForSelect(category, 'value')}),
+      ...(type && {type: flattenObjectForSelect(type, 'value')}),
+      ...(provider && {provider: flattenObjectForSelect(provider, 'value')}),
+      ...(references && {references: flattenArrayForSelect(references, 'id')}),
+    };
 
-      setIsLoading(false);
+    try {
+      setIsLoading && setIsLoading(true);
+      await handleLayerForm(false, parsed, id, selectedGroup);
+      const res = await getLayer(encodedQuery);
+      setLayer(res.data);
+      setIsLoading && setIsLoading(false);
+      setIsEditing && setIsEditing(false);
+    } catch (error) {
+      setIsLoading && setIsLoading(false);
+      setServerErrors && setServerErrors(error.data.errors);
     }
+  }
 
-    permissions && setupLayers();
-  }, [pageCursor, searchValue]);
-
-  return (
-    <LayerContext.Provider
-      value={{
-        handleSearchValueChange,
-        handleCursorChange,
-        isLoading,
-        isNoMore,
-        layers,
-        nextCursor,
-        totalResults,
-        pageSize,
-        searchValue,
-        selectedItem,
-      }}
-    >
-      <SidebarLayout page={PAGE_TYPE}>
-        <LayerList/>
-      </SidebarLayout>
-    </LayerContext.Provider>
-  );
-}
+  const handleJsonChange = (json) => {
+    try {
+      JSON.parse(json);
+    } catch (err) {
+      setJsonError(true);
+    }
+    if (!JSHINT.errors.length) {
+      const parsedJson = JSON.parse(json);
+      setLayerConfig(parsedJson);
+      setJsonError(false);
+      return parsedJson;
+    }
+    setJsonError(true);
+  };
 
 
-export function LayerDetail(path: any) {
-  const {selectedGroup} = useAuth0();
-  const encodedQuery = encodeQueryToURL(`layers/${path.page}`, {
-    ...LAYER_DETAIL_QUERY,
-    ...{group: selectedGroup},
-  });
+  function handleServerErrors(errors) {
+    setServerErrors(errors);
```

Comment:
`serverErrors` is not being rendered anywhere. Not sure where to put it, but it should at least exist